core/lib/spree/permission_sets/user_management.rb

```diff
@@ -2,7 +2,16 @@ module Spree
   module PermissionSets
     class UserManagement < PermissionSets::Base
       def activate!
-        can :manage, Spree.user_class
+        can [:admin, :display, :create, :update], Spree.user_class
+
+        # due to how cancancan filters by associations,
+        # we have to define this twice, once for `accessible_by`
+        can :update_email, Spree.user_class, spree_roles: { id: nil }
+        # and once for `can?`
+        can :update_email, Spree.user_class do |user|
+          user.spree_roles.none?
```

Comment:
This still feels weird to me since it assumes roles are only used for giving admin permissions and would work in unexpected ways if roles were used for other things.

Stembolt: Have you guys ever seen roles used for things other than admin permissions? (@gmacdougall / @cbrunsdon / @jhawthorn / @Senjai)